src/OpenColorIO/ops/OpArray.h

```diff
@@ -113,6 +115,12 @@ template<typename T> class ArrayT : public ArrayBase
             bool sameCoeff = true;
             for (unsigned long idx = 0; idx < m_length && sameCoeff; ++idx)
             {
+                if (IsNan(m_data[idx * 3]) &&
```

Comment:
Should this also handle the case where only one or two components are NaNs?

Reply:
Good question, but no.  If one component is NaN and another is not, it means the channels are not all equal.  Therefore, m_numColorComponents should remain 3.  So we want to do the next if-statement, set sameCoeff=false and then break.